kickstarter/android-oss
# NT-1387: Fix - Select AddOns for digital reward

# 📲 What

- A digital reward can have digital AddOns.

# 🤔 Why

- the current change is needed because we first assumed that AddOns where available just for shippable rewards.

# 🛠 How

- And empty shipping location has been created in case a digital reward is selected and it has add-Ons, doing so the extra logic needed for the AddOns screen is minimum
- Modified old test, and added new ones to cover this situation
- the RewardUtils.isDigital() method is now covering the logic for both models V1 and GraphQL rewards, before it was giving us false negatives, keep in mind the selected reward is from V1, the loaded addOns from GraphQL

# 👀 See

![select_digital_addOn](https://user-images.githubusercontent.com/4083656/92153126-9ebb1100-edd8-11ea-8b3e-4d23e6a60d8e.gif)

|  |  |

# 📋 QA

- Use any project with AddOns, this concrete one has a digital reward with digital addOns: [one inch straw - Loose reusable units - GROW IT from 1"- 10"](https://staging.kickstarter.com/projects/eserrano/one-inch-straw-loose-reusable-units-grow-it-from-1-10?ref=nav_search&result=project&term=one%20inch%20straw)
- Check we haven't introduce any regression for all the other use cases, this project has a bunch of AddOns to test with [GHOST BAND: A 96 page graphic novel performance!](https://staging.kickstarter.com/projects/ramonesome/ghost-band?ref=nav_search&result=project&term=ghost%20band)

# Story 📖

[Filter AddOns by location](https://kickstarter.atlassian.net/browse/NT-1387)
